linz.js

```diff
@@ -663,18 +645,35 @@ Linz.prototype.buildNavigation = function (cb) {
 
         },
 
-        function (done) {
+		// this function will run and return either a no-op function,
+		// or a function that will execute the 'navigation configuration' function and override
+		// nav with the customisation as returned to the callback
+		(function () {
```

Comment:
This is a rethink of an issue we keep coming up against. The issue is:

__*how best do we structure a series of async functions, in which not all of them are required under all conditions*__

This implementation uses a self executing function to determine if a function is required, or a no-op should be used in place of performing any actions...